tests/MongoDB.Driver.Tests/OcspIntegrationTests.cs

```diff
@@ -0,0 +1,144 @@
+         * the revoked certificate case */
+        private DisposableMongoClient CreateDisposableMongoClient(bool tlsInsecure)
+        {
+            var settings = DriverTestConfiguration.GetClientSettings().Clone();
+            settings.SslSettings.CheckCertificateRevocation = true;
+            // setting AllowInsecureTls= true will automatically set CheckCertificateRevocation to false
+            settings.AllowInsecureTls = tlsInsecure;
+            /* We want the heartbeat to fire so that we can get the HeartBeat exception in the cluster description
+             * in the exception that will be thrown when testing invalid certificates */
+            settings.HeartbeatInterval = TimeSpan.FromMilliseconds(500);
+            /* We lower the server selection timeout to speed up tests, but choosing a value <=5s may
+             * result in the driver being unable to perform OCSP endpoint checking in time, causing a
+             * ServerSelectionTimeout that does include a certificate revocation status error message. */
+            settings.ServerSelectionTimeout = TimeSpan.FromSeconds(5*2); // must be > 5s
+            // settings.SdamLogFilename = @"C:\temp\sdam" + $"{tlsInsecure}.log";
+
+            return new DisposableMongoClient(new MongoClient(settings));
+        }
+
+        private bool GetSecureClientExpectedOutcomeOrSkipTest()
+        {
+            var ocspOutcomeEnvironmentVariableValue
+                = Environment.GetEnvironmentVariable(_ocspOutcomeEnvironmentVariableName);
+            if (ocspOutcomeEnvironmentVariableValue == null)
+            {
+                throw new SkipException($"Skipping because {_ocspOutcomeEnvironmentVariableName} not set.");
+            }
+            if (!Boolean.TryParse(ocspOutcomeEnvironmentVariableValue, out var successExpected))
+            {
+               throw new Exception(
+                   $"Invalid value of {ocspOutcomeEnvironmentVariableValue} in {_ocspOutcomeEnvironmentVariableName}." +
+                   $" Expected true/false");
+            }
+            return successExpected;
+        }
+
+        /// <summary>
+        /// Sets up environment variables required for local testing.
+        /// </summary>
+        /// <param name="mockResponderValidatesAllCertificates">
+        /// Set to true if the mock OCSP responder will validate all certificates.
+        /// Set to false if the mock OCSP responder will revoke all certificates.
+        /// </param>
+        private void SetupLocalTestingEnvironmentVariables(bool mockResponderValidatesAllCertificates)
+        {
+            Environment.SetEnvironmentVariable("MONGODB_URI", "mongodb://localhost/?tls=true");
+            Environment.SetEnvironmentVariable(
+                _ocspOutcomeEnvironmentVariableName,
+                mockResponderValidatesAllCertificates.ToString());
+        }
+
+    }
+}
```

Comment:
This file has many more comments than we usually need.

When this many comments are needed it's usually a sign that the code is hard to understand.